src/emulator/auth/handlers.ts

```diff
@@ -36,6 +36,39 @@ export function registerHandlers(
     const state = getProjectStateByApiKey(apiKey);
 
     switch (mode) {
+      case "recoverEmail": {
+        const EXPIRED_LINK_ERROR = {
+          error: `Your request to revert your email has expired or the link has already been used.`,
+          instructions:
+            "Try reverting using this link again. If you're still unsuccessful. You can edit the email in the Emulator UI.",
+        };
+        const oob = state.validateOobCode(oobCode);
+        if (oob?.requestType !== "RECOVER_EMAIL") {
+          return res.status(400).json({
+            authEmulator: EXPIRED_LINK_ERROR,
```

Comment:
Let's give a more direct error message (wrong type of action code) here.

Reply:
Done.